### ISSUE TITLE
Type replacement by PostgreSQL domain

### ISSUE THREAD

**Issue by natsukagami** [volatiletech/sqlboiler#530]
In my application I use PostgreSQL's `domain`s to define types with check conditions. For example
```sql
CREATE DOMAIN uint256 AS numeric(78, 0)
	CHECK (
		value >= 0
			AND value < 115792089237316195423570985008687907853269984665640564039457584007913129639936);
```
and use this domain in a lot of columns across the database. 
When it comes to generating the SQL, I would like to map this into a `big.Int` (or a wrapper of it, written by me), but this is not doable without replacing the whole `numeric` type, but we also use the generic `numeric` type for other columns.

I propose we add a `DomainName` field into `drivers.Column`, or modify the postgreSQL driver to provide the domain name as part of an existing field (`FullDBType`...?)

I believe the practice of using domains for refined types is common, so this is a valuable feature to add. Thank you!

**Comment by aarondl** [comment]
I think the cleanest thing to do would be to add domain and allow type replacements on that field. I'd welcome a PR for this feature.

**Comment by aarondl** [comment]
I believe this is closed now thanks to #538
